packages/@storefront/structure/src/filtered-list/index.ts

```diff
@@ -33,14 +33,21 @@ class FilteredList {
     this.updateItems();
   }
 
-  onKeyDown(event) {
-    const value = this.refs.filter.value.trim().toLowerCase();
-    if (event.keyCode === RETURN_KEY_CODE) {
-      const foundItem = this.props.items.find((item) => item && (<FilteredList.ItemObject>item).value && (<FilteredList.ItemObject>item).value.toLowerCase() === value);
+  onKeyDown(event: KeyboardEvent & Tag.Event) {
+    if (event.keyCode !== RETURN_KEY_CODE) {
+      return;
+    }
 
-      if (foundItem && typeof (<FilteredList.ItemObject>foundItem).onClick === 'function') {
-        (<FilteredList.ItemObject>foundItem).onClick(event)
-      }
+    const value = this.refs.filter.value.trim().toLowerCase();
+    const foundItem = this.state.items.length === 1
+      ? this.state.items[0]
+      : this.props.items.find((el) => {
+        const item: FilteredList.ItemObject = <FilteredList.ItemObject>el;
+        return item && item.value && item.value.toLowerCase() === value;
+      });
+
+    if (foundItem && typeof (<FilteredList.ItemObject>foundItem).onClick === 'function') {
```

Comment:
Can we do the same annotate + cast trick for `foundItem`?

Reply:
This has been implemented. Resolving.